### ISSUE TITLE
Idea: Metadata on raw values

### ISSUE THREAD

**Issue by Georgantas** [mempool/mempool#1063]
### Description

![Screenshot from 2021-12-30 15-40-52](https://user-images.githubusercontent.com/18753033/147787299-8d263ee3-05e6-43d4-922b-b52c64371c1a.png)

Some ideas for the above screenshot:
1. In the witness, if the user hovers over the first value (`3045...`), they get a small dialog that indicates this value is the signature (like `SIGNATURE`). Additionally, maybe some info on what was signed to derive this value.
2. When the user hovers over the second value (`034db...`), they get a dialog that indicates this is the pubkey (like `PUBKEY`).
3. When the user hovers over the previous output script value (`555cc...`), they get a dialog that indicates this value is the HASH160 of the pubkey (like `HASH160(PUBKEY)`). Additionally, the dialog from idea 2 can also appear.
4. A "phantom" `CHECKSIG` operation as a third witness value to indicate this value is not actually there, but this is how P2WPKH is evaluated (per https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#p2wpkh).
5. Somehow express (possibly through a similar dialog) that the bech32 address (`bc1q24...`) is derived from the output script (`555cc...`).

Similar metadata dialogs could also be created for other address types (`P2WSH`, etc.).

Let me know what you think. Would these bloat the user experience, or confuse users?

**Comment by wiz** [comment]
Yeah, great suggestions, I also wanted to add more coloring into the raw bytes but your mouseover ideas make it even better

**Comment by Georgantas** [comment]
Thank you, free to tweak as you see fit :)

**Comment by antonilol** [comment]
just found this on the internet
https://nioctib.tech/#/transaction/f2f398dace996dab12e0cfb02fb0b59de0ef0398be393d90ebc8ab397550370b
would be cool to have something like this in mempool!

**Comment by Georgantas** [comment]
That's really cool! I love that I could step through the script execution: https://nioctib.tech/#/transaction/242513e8ac74052c339aa5265af843a5e45b34800b787782b0da582d595eacfa/input/0/interpret?step=0

**Comment by Georgantas** [comment]
> just found this on the internet https://nioctib.tech/#/transaction/f2f398dace996dab12e0cfb02fb0b59de0ef0398be393d90ebc8ab397550370b would be cool to have something like this in mempool! (click the raw ![image](https://user-images.githubusercontent.com/49868160/147889508-2c01ebb2-baf9-4a60-9bc1-63837556d101.png))

Maybe [this](https://mempool.space/api/tx/242513e8ac74052c339aa5265af843a5e45b34800b787782b0da582d595eacfa/hex) page could be enhanced with this coloring as well.

**Comment by antonilol** [comment]
> That's really cool! I love that I could step through the script execution: https://nioctib.tech/#/transaction/242513e8ac74052c339aa5265af843a5e45b34800b787782b0da582d595eacfa/input/0/interpret?step=0

yeah played with it too. would like to see that in mempool too but accept if thats too much for 2.4

**Comment by antonilol** [comment]
> > just found this on the internet https://nioctib.tech/#/transaction/f2f398dace996dab12e0cfb02fb0b59de0ef0398be393d90ebc8ab397550370b would be cool to have something like this in mempool! (click the raw ![image](https://user-images.githubusercontent.com/49868160/147889508-2c01ebb2-baf9-4a60-9bc1-63837556d101.png))
> 
> Maybe [this](https://mempool.space/api/tx/242513e8ac74052c339aa5265af843a5e45b34800b787782b0da582d595eacfa/hex) page could be enhanced with this coloring as well. Got to it by clicking the "Transaction Hex" link in the "Details" section of a transaction: ![Screenshot from 2022-01-02 16-36-54](https://user-images.githubusercontent.com/18753033/147890141-586360b0-8c46-4232-958b-41dcc041ba64.png)

A separate page for that would be needed. When you click that link for the TX hex you are actually going to the mempool API. Its not even HTML. Its bare `text/plain` that your browser renders in a nice monospace font because of that content type
